src/broadcast/setup.ts

```diff
@@ -30,14 +33,57 @@ export default function setupBroadcastIpc({
   let spectateWorker: SpectateWorker | undefined;
   let broadcastWorker: BroadcastWorker | undefined;
   let prefixOrdinal = 0;
+  let spectateIdleTimer: NodeJS.Timeout | undefined;
+
+  // Helper to start/reset the idle timeout (only when no active broadcasts)
+  const resetSpectateIdleTimeout = async () => {
+    // Clear any existing timer
+    if (spectateIdleTimer) {
+      clearTimeout(spectateIdleTimer);
+      spectateIdleTimer = undefined;
+    }
+
+    // Only start timeout if worker exists and has no active broadcasts
+    if (spectateWorker) {
+      const openBroadcasts = await spectateWorker.getOpenBroadcasts().catch(() => []);
+      if (openBroadcasts.length === 0) {
+        log.debug("Starting spectate idle timeout");
+        spectateIdleTimer = setTimeout(async () => {
+          log.debug("Spectate idle timeout reached, disconnecting and terminating worker");
+          if (spectateWorker) {
+            await spectateWorker.disconnect();
+            await spectateWorker.terminate();
+            spectateWorker = undefined;
+          }
+        }, SPECTATE_IDLE_TIMEOUT);
+      }
+    }
+  };
+
+  // Helper to stop the idle timeout (when actively watching)
+  const stopSpectateIdleTimeout = () => {
+    if (spectateIdleTimer) {
+      log.debug("Stopping spectate idle timeout (actively watching)");
+      clearTimeout(spectateIdleTimer);
+      spectateIdleTimer = undefined;
+    }
+  };
 
   dolphinManager.events
     .filter<DolphinPlaybackClosedEvent>((event) => {
       return event.type === DolphinEventType.CLOSED && event.dolphinType === DolphinLaunchType.PLAYBACK;
     })
-    .subscribe((event) => {
+    .subscribe(async (event) => {
       if (spectateWorker) {
-        void spectateWorker.dolphinClosed(event.instanceId).catch(log.error);
+        await spectateWorker.dolphinClosed(event.instanceId).catch(log.error);
+
+        // Check if there are any remaining open broadcasts
+        const openBroadcasts = await spectateWorker.getOpenBroadcasts().catch(() => []);
+        if (openBroadcasts.length === 0) {
```

Comment:
I believe this check is redundant as `resetSpectateIdleTimeout()` already does it?